zeebe/gateway-rest/src/test/java/io/camunda/zeebe/gateway/rest/controller/JobQueryControllerTest.java

```diff
@@ -0,0 +1,295 @@
+                    "tenantId": "<default>"
+                }
+            ],
+            "page": {
+                "totalItems": 1,
+                "firstSortValues": ["f"],
+                "lastSortValues": ["v"]
+            }
+        }
+        """;
+  private static final SearchQueryResult<JobEntity> SEARCH_QUERY_RESULT =
+      new SearchQueryResult.Builder<JobEntity>()
+          .total(1L)
+          .items(
+              List.of(
+                  new JobEntity(
+                      1L,
+                      "testJob",
+                      "testWorker",
+                      JobState.COMPLETED,
+                      JobKind.TASK_LISTENER,
+                      ListenerEventType.COMPLETING,
+                      3,
+                      true,
+                      "test denied reason",
+                      false,
+                      "123",
+                      "test error message",
+                      Map.of("foo", "bar"),
+                      OffsetDateTime.parse("2025-06-05T09:05:00.000Z"),
+                      OffsetDateTime.parse("2025-06-05T10:05:00.000Z"),
+                      "processDefinitionId",
+                      2L,
+                      3L,
+                      "elementId",
+                      4L,
+                      "<default>")))
+          .firstSortValues(new Object[] {"f"})
+          .lastSortValues(new Object[] {"v"})
+          .build();
+  @MockBean JobServices<JobActivationResult> jobServices;
+  @MockBean MultiTenancyConfiguration multiTenancyCfg;
+  @MockBean ResponseObserverProvider responseObserverProvider;
+
+  @BeforeEach
+  void setupJobServices() {
+    when(jobServices.withAuthentication(ArgumentMatchers.any(Authentication.class)))
+        .thenReturn(jobServices);
+  }
+
+  @Test
+  void shouldSearchJobWithEmptyBody() {
+    // given
+    when(jobServices.search(any(JobQuery.class))).thenReturn(SEARCH_QUERY_RESULT);
+
+    // when / then
+    webClient
+        .post()
+        .uri(JOB_SEARCH_URL)
+        .exchange()
+        .expectStatus()
+        .isOk()
+        .expectHeader()
+        .contentType(MediaType.APPLICATION_JSON)
+        .expectBody()
+        .json(EXPECTED_SEARCH_RESPONSE);
+
+    verify(jobServices).search(new JobQuery.Builder().build());
+  }
+
+  @Test
+  void shouldSearchJobWithAllFilters() {
+    // given
+    when(jobServices.search(any(JobQuery.class))).thenReturn(SEARCH_QUERY_RESULT);
+
+    final var request =
+        """
+      {
+      "filter": {
+        "jobKey": 1,
+        "type": "testJob",
+        "worker": "testWorker",
+        "state": "COMPLETED",
+        "kind": "TASK_LISTENER",
+        "listenerEventType": "COMPLETING",
+        "processDefinitionId": "processDefinitionId",
+        "processDefinitionKey": 2,
+        "processInstanceKey": 3,
+        "elementId": "elementId",
+        "elementInstanceKey": 4,
+        "tenantId": "<default>"
+            }
+          }
+        }
```

Comment:
🙃 The formatting looks a bit off. Not a big deal but maybe we can align this a bit 🙂